src/ILVerify/src/ILImporter.StackValue.cs

```diff
@@ -92,19 +105,15 @@ static public StackValue CreateFromType(TypeDesc type)
                 case TypeFlags.UInt16:
                 case TypeFlags.Int32:
                 case TypeFlags.UInt32:
-                    return CreatePrimitive(StackValueKind.Int32);
                 case TypeFlags.Int64:
                 case TypeFlags.UInt64:
-                    return CreatePrimitive(StackValueKind.Int64);
                 case TypeFlags.Single:
                 case TypeFlags.Double:
-                    return CreatePrimitive(StackValueKind.Float);
                 case TypeFlags.IntPtr:
                 case TypeFlags.UIntPtr:
                 case TypeFlags.Pointer:
-                    return CreatePrimitive(StackValueKind.NativeInt);
                 case TypeFlags.Enum:
-                    return CreateFromType(type.UnderlyingType);
+                    return CreatePrimitive(type);
                 case TypeFlags.ByRef:
                     return CreateByRef(((ByRefType)type).ParameterType);
```

Comment:
Would it make sense to normalize the type to intermediate type here so that that StackValue is always intermediate type? It is mix now - it is normalized to intermediate type for primitives, but kept as actual type for byrefs.

Reply:
Yes, that is right. I forgot that ByRefs also have a different intermediate type, since the intermediate type is dependent on the verification type.
It does indeed make more sense to normalize to intermediate type in this method.